File: ios/BasicQRScanner/VisionKit/VisionKitViewManager.m
//
//  VisionKitViewManager.m
//  BasicQRApp
//
//  Created by Ananth Desai on 31/12/24.
//

#import <Foundation/Foundation.h>
#import "React/RCTViewManager.h"

@interface RCT_EXTERN_MODULE(VisionKitViewManager, RCTViewManager)
  RCT_EXPORT_VIEW_PROPERTY(onSuccessfulScan, RCTDirectEventBlock)
@end

File: ios/BasicQRScanner/VisionKit/VisionKitViewManager.swift
//
//  VisionKitViewManager.swift
//  BasicQRApp
//
//  Created by Ananth Desai on 31/12/24.
//

import Foundation

@available(iOS 16.0, *)
@objc(VisionKitViewManager)
class VisionKitViewManager: RCTViewManager {
  private weak var scannerRef: VisionKitView? = nil
  
  override class func requiresMainQueueSetup() -> Bool {
    false
  }
  
  override func view() -> UIView! {
    let scannerRef = VisionKitView()
    self.scannerRef = scannerRef
    return scannerRef
  }
}
